lib/data/table.js

```diff
@@ -1,14 +1,16 @@
+const DataTableArgument = require("./dataTableArgument");
+
 /**
  * Datatable class to provide data driven testing
  */
 class DataTable {
-  /** @param {Array<*>} array */
+  /** @param {Array<DataTableArgument>} array */
   constructor(array) {
```

Comment:
I'm not sure this is correct
As you can actually pass any array into DataTable
DataTableArgument works only with Gherkin tables while DataTable works with classical tables as well